client/client.go

```diff
@@ -1861,58 +1861,132 @@ func (c *Client) emitStats() {
 	}
 }
 
-// emitHostStats pushes host resource usage stats to remote metrics collection sinks
-func (c *Client) emitHostStats(hStats *stats.HostStats) {
+func (c *Client) setGaugeForMemoryStats() {
 	nodeID := c.Node().ID
-	metrics.SetGauge([]string{"client", "host", "memory", nodeID, "total"}, float32(hStats.Memory.Total))
-	metrics.SetGauge([]string{"client", "host", "memory", nodeID, "available"}, float32(hStats.Memory.Available))
-	metrics.SetGauge([]string{"client", "host", "memory", nodeID, "used"}, float32(hStats.Memory.Used))
-	metrics.SetGauge([]string{"client", "host", "memory", nodeID, "free"}, float32(hStats.Memory.Free))
+	hStats := c.hostStatsCollector.Stats()
```

Comment:
Call these in the emitHostStats and pass it in to all the methods?

Reply:
I was considering creating a separate ClientMetrics object and setting hStats, nodeID, etc as fields. What do you think about that, or should this logic be kept in Client?

Reply:
Maybe if we find that we add more than just the stats and nodeID but for just those two I don't think it is worth it.

Reply:
Base labels could also be a field- but agree, we can not do it for now and keep it as an option for later.